PWGHF/D2H/TableProducer/dataCreatorDplusPiReduced.cxx

```diff
@@ -75,16 +82,20 @@ struct HfDataCreatorDplusPiReduced {
   Configurable<std::string> ccdbPathGrp{"ccdbPathGrp", "GLO/GRP/GRP", "Path of the grp file (Run 2)"};
   Configurable<std::string> ccdbPathGrpMag{"ccdbPathGrpMag", "GLO/Config/GRPMagField", "CCDB path of the GRPMagField object (Run 3)"};
 
+  // CCDB service
   Service<o2::ccdb::BasicCCDBManager> ccdb;
   o2::base::MatLayerCylSet* lut;
   o2::base::Propagator::MatCorrType matCorr = o2::base::Propagator::MatCorrType::USEMatCorrLUT;
   int runNumber;
 
-  double massPi = RecoDecay::getMassPDG(kPiPlus);
-  double massD = RecoDecay::getMassPDG(pdg::Code::kDMinus);
-  double massB0 = RecoDecay::getMassPDG(pdg::Code::kB0);
-  double massDPi{0.};
-  double invMassD{0.};
+  // O2DatabasePDG service
+  Service<o2::framework::O2DatabasePDG> pdg;
```

Comment:
Here you define this `pdg` object because later you need it to get the particle masses from PDG. But where do you define it? You declare it here w/o definition, and then you de-reference the pointer in lines 155, 156, 157. Am I overlooking something?

Reply:
Actually, this is the first time I use this new functionality. I just followed the [tutorial](https://github.com/AliceO2Group/O2Physics/blob/master/Tutorials/src/usingPDGService.cxx#L20) indicated in the error messages of the log file.

Reply:
Thanks for providing the link to the tutorial. Yes, in principle the default constructor is called (see here: https://github.com/AliceO2Group/AliceO2/blob/dev/Framework/PhysicsSupport/include/Framework/O2DatabasePDGPlugin.h#L35C21-L35C42). Thank you.